sys-apps/less/files/lesspipe-r3.sh

```diff
@@ -0,0 +1,338 @@
+#!/bin/bash
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# Preprocessor for 'less'. Used when this environment variable is set:
+# LESSOPEN="|lesspipe %s"
+
+# TODO: handle compressed files better
+
+[[ -n ${LESSDEBUG+set} ]] && set -x
+
+trap 'exit 0' PIPE
+
+guesscompress() {
+	case "$1" in
+		*.gz|*.z)   echo "gunzip -c" ;;
+		*.bz2|*.bz) echo "bunzip2 -c" ;;
+		*.lz)       echo "lzip -dc" ;;
+		*.lzma)     echo "unlzma -c" ;;
+		*.lzo)      echo "lzop -dc" ;;
+		*.xz)       echo "xzdec" ;;
+		*.zst)      echo "zstdcat" ;;
+		*)          echo "cat" ;;
+	esac
+}
+
+crl_filter() {
+    if command -v certtool &>/dev/null; then
+        certtool --crl-info --text --infile "$1"
+    else
+        openssl crl -hash -text -noout -in "$1"
+    fi
+}
+
+csr_filter() {
+    if command -v certtool &>/dev/null; then
+        certtool --crq-info --text --infile "$1"
+    else
+        openssl req -text -noout -in "$1"
+    fi
+}
+
+crt_filter() {
+    if command -v certtool &>/dev/null; then
+        certtool --certificate-info --text --infile "$1"
+    else
+        openssl x509 -hash -text -noout -in "$1"
+    fi
+}
+
+jks_filter() {
+    if command -v keytool &>/dev/null; then
+        keytool -list -keystore "$1"
+    else
+        cat "$1"
+    fi
+}
+
+p12_filter() {
+    openssl pkcs12 -nokeys -info -in "$1"
+}
+
+lesspipe_file() {
+	local out=$(file -L -- "$1")
+	local suffix
+	case ${out} in
+		*" 7-zip archive"*) suffix="7z";;
+		*" ar archive"*)    suffix="a";;
+		*" CAB-Installer"*) suffix="cab";;
+		*" cpio archive"*)  suffix="cpio";;
+		*" ELF "*)          suffix="elf";;
+		*" LHa"*archive*)   suffix="lha";;
+		*" troff "*)        suffix="man";;
+		*" script text"*)   suffix="sh";;
+		*" shared object"*) suffix="so";;
+		*" tar archive"*)   suffix="tar";;
+		*" Zip archive"*)   suffix="zip";;
+		*": data")          hexdump -C -- "$1"; return 0;;
+		*)                  return 1;;
+	esac
+	lesspipe "$1" ".${suffix}"
+	return 0
+}
+
+lesspipe() {
+	local match=$2
+	[[ -z ${match} ]] && match=$1
+
+	local DECOMPRESSOR=$(guesscompress "${match}")
+
+	# User filters
+	if [[ -x ~/.lessfilter ]] ; then
+		~/.lessfilter "$1" && exit 0
+	fi
+
+	# System filters
+	if compgen -G "/etc/lessfilter.d/*.sh"; then
```

Comment:
`compgen` requires `RDEPEND=app-shells/bash[readline]`, see https://bugs.gentoo.org/909163. But maybe we can avoid using `compgen` here?

Reply:
I would suggest removing the `readline` use flag from the `gentoo/stage3` container to make this "issue" more obvious.

Using nullglob instead, not testing/refactoring the whole script so it's localized to just this change.

```diff
$ git show
commit 5e16f0f7562bc7405fc08bea4b3c803bdfb709cb (HEAD -> sys-apps_less_lesspipe)
Author: Victor Payno <vpayno@users.noreply.github.com>
Date:   Fri Dec 1 12:50:34 2023 -0800

    fixup! sys-apps/less: add 643-r1 (add certtool filters and system filter support)

diff --git a/sys-apps/less/files/lesspipe-r3.sh b/sys-apps/less/files/lesspipe-r3.sh
index 0bffa3313e0a..24166064425d 100644
--- a/sys-apps/less/files/lesspipe-r3.sh
+++ b/sys-apps/less/files/lesspipe-r3.sh
@@ -94,13 +94,13 @@ lesspipe() {
        fi
 
        # System filters
-       if compgen -G "/etc/lessfilter.d/*.sh"; then
-               for f in /etc/lessfilter.d/*.sh; do
-                       if [[ -x ${f} ]]; then
-                               "${f}" "$1" && exit 0
-                       fi
-               done
-       fi
+       shopt -s nullglob
+       for f in /usr/lib/lessfilter.d/*; do
+               if [[ -x ${f} ]]; then
+                       "${f}" "$1" && exit 0
+               fi
+       done
+       shopt -u nullglob
 
        local ignore
        for ignore in ${LESSIGNORE} ; do
```